### ISSUE TITLE
Why predict one point (xmin,ymin = xmax,ymax) in other picture?

### ISSUE THREAD

**Issue by QiaoranC** [zisianw/FaceBoxes.PyTorch#5]
Hi:
i am trying to do an inference in my picture(not from dataset), the image preprocess (orginal size input, not resize), model  load, results post process is basically follow your test.py.  
But the predict point is funny, which shows following:
**there just one point, xmin,ymin equal xmax ymax**

`dets 
[[2.1120000e+03 1.3440000e+03 2.1120000e+03 1.3440000e+03 9.8217058e-01]]
`

Just curious, Any ideal why this happen?

So far, there are two thing i noticed:
1, the predict point is closer to the centre point of the box that mtcnn predict, Very Strange.
facebox: [2240,960,2240,960]
mtcnn: [1626,503,2737,1882] 

2, the orginal predict is just one point
because i trace every step, found in decode step, before `boxes[:, :2] -= boxes[:, 2:] / 2`,  boxes was like [0.x,0.x,0,0], the last two is zero.

```
def decode(loc, priors, variances):
   boxes = torch.cat((
        priors[:, :2] + loc[:, :2] * variances[0] * priors[:, 2:],
        priors[:, 2:] * torch.exp(loc[:, 2:] * variances[1])), 1)
    boxes[:, :2] -= boxes[:, 2:] / 2
    boxes[:, 2:] += boxes[:, :2]
    return boxes
```

**Comment by hsl20130659** [comment]
编译nms，cpu和gpu编译都不通过，你是怎么编译的呢

**Comment by QiaoranC** [comment]
> 编译nms，cpu和gpu编译都不通过，你是怎么编译的呢

我没用那个nms

**Comment by zisianw** [comment]
@QiaoranC Hi, I have not faced this situation before. Can you try on some public images and see if they would generate a similar result as yours? You can post the link to the public image here if it's the case.

**Comment by QiaoranC** [comment]
@zisianw 
hi, i chose a public image [link](https://previews.123rf.com/images/subbotina/subbotina1510/subbotina151000001/46048732-beautiful-spa-model-girl-with-perfect-fresh-clean-skin.jpg), give same one point results. 

`xmin 704.0 ymin 576.2 xmax 704.0 ymax 576.0
`

 if it helps. this is my scrip, a simple directly inference:
ignore the load_model function part same from your test.py

```
import torch
import torch.nn as nn
import torchvision.transforms as transforms
import torchvision
from itertools import product as product

import cv2
import numpy as np
from PIL import Image

from faceboxes import FaceBoxes
from layers.functions.prior_box import PriorBox
from utils.box_utils import decode

trained_model_path = './face_box/FaceBoxes.pth'


model = FaceBoxes(phase='test', size=None, num_classes=2)
device = torch.device("cuda")
model = load_model(model, trained_model_path) 
model = model.to(device)
model.eval()

image_path = 'test.jpg'
img = np.float32(cv2.imread(image_path, cv2.IMREAD_COLOR))
im_height, im_width, _ = img.shape

scale = torch.Tensor([img.shape[1], img.shape[0], img.shape[1], img.shape[0]])
img -= (104, 117, 123)
img = img.transpose(2, 0, 1)
img = torch.from_numpy(img).unsqueeze(0)
# if args.cuda:
img = img.cuda()
scale = scale.cuda()

loc, conf, _ = out
prior_data = priors.data
boxes = decode(loc.data.squeeze(0), prior_data, cfg['variance'])
boxes = boxes * scale
boxes = boxes.cpu().numpy()
scores = conf.data.cpu().numpy()[:, 1]

# ignore low scores
confidence_threshold = 0.9
inds = np.where(scores > confidence_threshold)[0]
boxes = boxes[inds]
scores = scores[inds]
# print('boxes',boxes)

# keep top-K before NMS
order = scores.argsort()[::-1][:5]
boxes = boxes[order]
scores = scores[order]

# print('boxes_list',boxes_list)
dets = np.hstack((boxes, scores[:, np.newaxis])).astype(np.float32, copy=False)
print('dets',dets)
dets = dets[:5, :]
print('dets[:5, :]',dets)
for k in range(dets.shape[0]):
    xmin = dets[k, 0]
    ymin = dets[k, 1]
    xmax = dets[k, 2]
    ymax = dets[k, 3]
    ymin += 0.2 * (ymax - ymin + 1)
    score = dets[k, 4]
print('xmin {:.1f} ymin {:.1f} xmax {:.1f} ymax {:.1f}\n'.format(xmin, ymin, xmax, ymax))
```

**Comment by zisianw** [comment]
Hi @QiaoranC, I'm not sure of the modification you have done.. but by changing the input image to the one you provided in the test code. I have the following results:
```
dets:
[[4.7637433e+02 1.4522061e+02 8.7534143e+02 7.0209723e+02 9.9999714e-01]
 [7.2682770e+01 4.4329388e+02 1.3906261e+02 5.2275232e+02 6.7398451e-02]
 [1.8745308e+02 9.9066504e+02 7.2990295e+02 1.5005250e+03 5.6240465e-02]
 [1.1730062e+03 8.2211053e+02 1.2306016e+03 8.8890796e+02 5.0705820e-02]]
```
The first bounding box detects the face in the image.

**Comment by QiaoranC** [comment]
@zisianw Thanks for the test, there are some slightly environment difference, like i using pytorch 0.4.1, did not run mask.sh, but i dont think that really matter in this case.
Surprisingly (maybe not),  i can reproduce the results in same script in python3 (using python2 before).
I am interest in why python2 make such a big difference, i will continue looking for details. 
Thanks.

**Comment by zisianw** [comment]
@QiaoranC If it is due to python2, it might be how division is handled differently in python2 and python3. You could update us if you find anything interesting :) cheers.

**Comment by QiaoranC** [comment]
@zisianw 
indeed it is the division problem, add `from __future__ import print_function, division` in `PriorBox.py in layers.functions.prior_box` will solve it.
now i can formally compare this model with mtcnn (especially in speed, accuracy, failure rate(not found face) , look forward a better results. 
And thanks for your work. face detect may one of the most common process in some area. 
Any noticeable improvement will appreciate it.
